src/builtins/string.cpp

```diff
@@ -735,6 +736,8 @@ static int string_join_maybe0(parser_t &parser, io_streams_t &streams, int argc,
     arg_iterator_t aiter(argv, optind, streams);
     while (const wcstring *arg = aiter.nextstr()) {
         if (!opts.quiet) {
+            if (opts.no_empty && arg->length() == 0) continue;
```

Comment:
`arg->empty()` or clang-tidy will yell at us (yes that's kind of useless churn, sorry)

Reply:
Have a look at [what actually failed](https://github.com/fish-shell/fish-shell/runs/5501523442?check_suite_focus=true) - the test was flaky, it had nothing to do with this code. It didn't even execute it.

Common CI systems are typically very very resource-constrained and we have a few tests that involve some form of timing, so they do, unfortunately, occasionally fail.

There's no reason to revert this.